server/src/main/java/dev/slimevr/vr/processor/skeleton/TapDetection.java

```diff
@@ -0,0 +1,225 @@
+package dev.slimevr.vr.processor.skeleton;
+
+import java.util.LinkedList;
+import com.jme3.math.Vector3f;
+
+import dev.slimevr.config.TapDetectionConfig;
+import dev.slimevr.osc.VRCOSCHandler;
+import dev.slimevr.vr.trackers.Tracker;
+
+
+// class that monitors the acceleration of the waist, hip, or chest trackers to detect taps
+// and use this to trigger a quick reset (if your wondering why no single tap class exists, it's because 
+// to many false positives)
+public class TapDetection {
+
+	// server and related classes
+	private HumanSkeleton skeleton;
+	private VRCOSCHandler oscHandler;
+	private TapDetectionConfig config;
+
+
+	// tap detection
+	private boolean enabled = false;
+	private LinkedList<float[]> accelList = new LinkedList<>();
+	private LinkedList<Float> tapTimes = new LinkedList<>();
+	private float resetDelayNs = 0.20f * 1000000000.0f;
+
+	// hyperparameters
+	private static final float NEEDED_ACCEL_DELTA = 6.0f;
+	private static final float ALLOWED_BODY_ACCEL = 1.5f;
+	private static final float ALLOWED_BODY_ACCEL_SQUARED = ALLOWED_BODY_ACCEL * ALLOWED_BODY_ACCEL;
+	private static final float CLUMP_TIME_NS = 0.03f * 1000000000.0f;
+	private static final float TIME_WINDOW_NS = 0.5f * 1000000000.0f;
+
+	// state
+	private float resetRequestTime = -1.0f;
+
+	public TapDetection(HumanSkeleton skeleton) {
+		this.skeleton = skeleton;
+	}
+
+	public TapDetection(
+		HumanSkeleton skeleton,
+		VRCOSCHandler oscHandler,
+		TapDetectionConfig config
+	) {
+		this.skeleton = skeleton;
+		this.oscHandler = oscHandler;
+		this.config = config;
+		updateConfig();
+	}
+
+	public void updateConfig() {
+		this.enabled = config.getEnabled();
+		this.resetDelayNs = config.getDelay() * 1000000000.0f;
+	}
+
+	public void setEnabled(boolean enabled) {
+		this.enabled = enabled;
+	}
+
+	public boolean getEnabled() {
+		return enabled;
+	}
+
+	// main function for tap detection
+	public void update() {
+		if (skeleton == null || !enabled)
+			return;
+
+		Tracker tracker = getTrackerToWatch();
+		if (tracker == null)
+			return;
+
+		// check if we should reset
+		if (resetRequestTime != -1.0f) {
+			if (System.nanoTime() - resetRequestTime > resetDelayNs && oscHandler != null) {
+				oscHandler.yawAlign();
+				skeleton.resetTrackersYaw();
+				resetRequestTime = -1.0f;
+			}
```

Comment:
This feature won't reset without OSC? Maybe the null check should be around only the `oscHandler` part...
```suggestion
			if (System.nanoTime() - resetRequestTime > resetDelayNs) {
				if (oscHandler != null)
					oscHandler.yawAlign();
				skeleton.resetTrackersYaw();
				resetRequestTime = -1.0f;
			}
```